### PR TITLE
lesson1 homework

### PR DESCRIPTION
**Pull request by kristinaChander**

### CODE REVIEW

File: src/test/resources/hw1/AllTestsSuite.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        <classes>
+            <class name="hw1.CalculatorSumTest"/>
+        </classes>
+    </test>
+
+    <test name="Subtract test">
+        <classes>
+            <class name="hw1.CalculatorSubtractTest"/>
+        </classes>
+    </test>
```
Comment (DmitryKhodakovsky) [inline src/test/resources/hw1/AllTestsSuite.xml]: Why do you decide include all test classes into this suite? Do we have other way run all tests in the suite?

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]
Please fix comments.
When I try to run test on my laptop with command `mvn clean test` than I've got an follow error:

```
EPRUPETW2288:KristinaChander dmitry_khodakovsky$ mvn clean test
[INFO] Scanning for projects...
[INFO] 
[INFO] -------------------------< com.epam:test-auto >-------------------------
[INFO] Building test-auto 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for com.epam.tat.module4:calculator:jar:1.0 is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.234 s
[INFO] Finished at: 2020-03-27T23:22:13+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project test-auto: Could not resolve dependencies for project com.epam:test-auto:jar:1.0-SNAPSHOT: Failure to find com.epam.tat.module4:calculator:jar:1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

File: src/test/resources/hw1/AllTestsSuite.xml (hunk 1 above)
Comment (kristinaChander) [reply src/test/resources/hw1/AllTestsSuite.xml]: We may use a package. Then all tests inside the package will run.

File: src/test/java/hw1/CalculatorDivideTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        };
+    }
+
+    @DataProvider
+    public static Object[][] divideByZeroLongArgumentsTestData() {
+        return new Object[][]{
+                {1},
+                {0}
+        };
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw1/CalculatorDivideTest.java]: Why *DataProviders* have `static` modifier?

Please check other test classes

File: src/test/java/hw1/CalculatorDivideTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import static org.testng.Assert.assertEquals;
+
+public class CalculatorDivideTest {
+
+    private Calculator calculator;
+
+    @BeforeMethod
+    public void setCalculator() {
+        calculator = new Calculator();
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw1/CalculatorDivideTest.java]: I could see in the following classes code duplication: 
- CalculatorDivideTest
- CalculatorMultiplyTest
- CalculatorSubtractTest
- CalculatorSumTest

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]
Please resolve comments

**Review by DmitryKhodakovsky — APPROVED** [review]
HW1 is accepted. could be merged to the master branch

**Pull request merged**